snapcraft/services/package.py

```diff
@@ -122,7 +122,10 @@ def write_metadata(self, path: pathlib.Path) -> None:
             manifest = lifecycle.generate_manifest()
             manifest.to_yaml_file(snap_dir / "manifest.yaml")
 
-            shutil.copy(self._snapcraft_yaml_path, snap_dir)
+            try:
+                shutil.copy(self._snapcraft_yaml_path, snap_dir)
+            except shutil.SameFileError:
+                pass
```

Comment:
Maybe try this instead
```suggestion
            _hardlink_or_copy(self._snapcraft_yaml_path, snap_dir / "snapcraft.yaml")
```

Reply:
I updated the PR description to help understand. And the `_hardlink_or_copy` will delete the file we want, since it is the same file and path.